app/src/main/java/com/devshish/internship/presentation/ui/search/SearchViewModel.kt

```diff
@@ -0,0 +1,28 @@
+package com.devshish.internship.presentation.ui.search
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.devshish.internship.data.repository.SearchAPIRepository
+import com.devshish.internship.domain.model.searchDTO.HitDTO
+import com.devshish.internship.domain.repository.ISearchApiRepository
+import kotlinx.coroutines.launch
+
+class SearchViewModel(
+    private val repository: SearchAPIRepository
+) : ViewModel() {
+
+    val searching: LiveData<List<HitDTO>>
+        get() = _searching
+    private val _searching = MutableLiveData<List<HitDTO>>()
+
+    fun searchSongs(query: String) {
+        viewModelScope.launch {
+            val response = repository.searchSongs(query)
+            if (response.isSuccessful) {
+                _searching.value = response.body()?.response?.hits
+            }
+        }
+    }
```

Comment:
For now, remove Response, and just use try/catch (add logging if error occurs)

Reply:
Also, I saw that you show/hilde loading in fragment, but it will be cooler and more correct to store in viewModel isLoading : LiveData<Boolean> , and rule state from viewmodel, not from fragment. 
Fragment will just observe loading state and show/hide it